templates/periodicPayment.go

```diff
@@ -0,0 +1,108 @@
+package templates
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// PeriodicPayment template representation
+type PeriodicPayment struct {
+	ContractTemplate
+	amount      uint64
+	receiver    types.Address
+	leaseBase64 string
+	period      uint64
+	duration    uint64
+}
+
+// GetWithdrawalTransaction returns a signed transaction extracting funds from the contract
+// fee: the fee to pay in microAlgos
+// firstValid: the first round on which the txn will be valid
+// genesisHash: the hash representing the network for the txn
+func (c PeriodicPayment) GetWithdrawalTransaction(fee, firstValid uint64, genesisHash []byte) ([]byte, error) {
```

Comment:
Same like the others. Does the function expect the user to re-instantiate the contract?